hostpool.go

```diff
@@ -178,6 +236,9 @@ func (p *standardHostPool) setHosts(hosts []string) {
 			host:       h,
 			retryDelay: p.initialRetryDelay,
 		}
+		if p.maxFailures > 0 {
+			e.failures = NewRingBuffer(p.maxFailures + 1)
```

Comment:
is it the case we use `p.maxFailures + 1` here because we want to say something like "remove host after 3 errors", which means we need to know when we get the fourth error?

Reply:
That's right -- I lost the commentary in a refactor. Will add it back in.